0121-forms-pattern-design-guidance.md

```diff
@@ -0,0 +1,46 @@
+---
+category: pattern design guidance
+state: draft
+created: 2024-10-16
+updated: 2024-10-22
+---
+
+# Forms Pattern Design Guidance
+
+A [form](https://clarity.design/documentation/forms) is a collection of related input controls that allows a user to provide data or configure options. Forms are composed of control components such as [checkboxes](https://clarity.design/documentation/checkbox), [dropdowns](https://clarity.design/documentation/dropdown), [input fields](https://clarity.design/documentation/input), [radio buttons](https://clarity.design/documentation/radio), [toggle switches](https://clarity.design/documentation/toggle-switch) and more.
+
+## Guidance
+
+- **Must** be clear and simple. This will increase the quality of the response from the user and help prevent incorrect data submission.
+- **Must** include controls that are required. Review each component and eliminate unnecessary ones.
+- **Must** add an asterisk after the label for required fields and include a legend at the top of the form to inform users that the asterisk denotes a required field unless it's obvious that the fields are required to proceed. An example exception of an exception to this rule is a login form.
+- **Must not** place interactive content (buttons, links, etc) inside the label.
```

Comment:
```suggestion
- Design teams **must not** place interactive content, such as buttons or links, inside the label, as this can confuse users and hinder usability.
```